test/functionalTest/cases/4004_jexl_expressions_in_subs/jexl_transformation_full.test

```diff
@@ -0,0 +1,450 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+# JEXL_EXPR_FLAVOUR - to mark the test has to execute only when contextBroker includes jexl-expr flavour
+
+--NAME--
+JEXL expression in custom notification (using all transformations)
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create custom sub using all transformations
+# 02. Create entity E1 with A to V attributes
+# 03. Dump accumulator and see expected notification
+#
+
+
+echo "01. Create custom sub using all transformations"
+echo "==============================================="
+# NOTE: '\'' is the way of scaping a ' in the payload variable below (see https://stackoverflow.com/a/1250279/1485926)
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "idPattern": ".*",
+        "type": "T"
+      }
+    ]
+  },
+  "notification": {
+    "attrs": [
+      "A",
+      "B",
+      "C",
+      "D",
+      "E",
+      "F",
+      "G",
+      "H",
+      "I",
+      "J",
+      "K",
+      "L",
+      "M",
+      "N",
+      "O",
+      "P",
+      "Q",
+      "R",
+      "S",
+      "T",
+      "U",
+      "V"
+    ],
+    "httpCustom": {
+      "url": "http://127.0.0.1:'${LISTENER_PORT}'/notify",
+      "ngsi": {
+        "A": {
+          "type": "Text",
+          "value": "${A+(3|toString)}"
+        },
+        "B": {
+          "type": "Text",
+          "value": "${B|replaceStr('\''A'\'','\''B'\'')}"
+        },
+        "C": {
+          "type": "Number",
+          "value": "${C+1}"
+        },
+        "D": {
+          "type": "Text",
+          "value": "${D|uppercase}"
+        },
+        "E": {
+          "type": "TextUnrestricted",
+          "value": "${E|lowercase}"
+        },
+        "F": {
+          "type": "Text",
+          "value": "${F[1]|trim}"
+        },
+        "G": {
+          "type": "Number",
+          "value": "${G|toFixed(1)}"
+        },
+        "H": {
+          "type": "Text",
+          "value": "${H|includes('\''N'\'')}"
+        },
+        "I": {
+          "type": "Text",
+          "value": "${I|indexOf('\''test'\'')}"
+        },
+        "J": {
+          "type": "Text",
+          "value": "${(J|isNaN)}"
+        },
+        "K": {
+          "type": "Text",
+          "value": "${K|typeOf}"
+        },
+        "L": {
+          "type": "TextUnrestricted",
+          "value": "${(L|split('\'' '\''))}"
+        },
+        "M": {
+          "type": "Number",
+          "value": "${M|round(2)}"
```

Comment:
`round` does not receive arguments, is it this expression intentional?

Reply:
No, it's a typo. Let's fix it an check the test is still passing

```suggestion
          "value": "${M|round}"
```

(Waiting to finish the current test pass to commit this suggestion...)

Reply:
Fixed in [cfa543a](https://github.com/telefonicaid/fiware-orion/pull/4511/commits/cfa543a5408363e6be32895bd27fe4b5d21a8d68)